CONTRIBUTING.md

```diff
@@ -16,48 +16,73 @@ We use `release` as our release branch. _Generally, changes should never go dire
+3. When entering a new issue, please include all relevant information. For content issues, include
+   the book or books this applies to, and specific locations that should be reviewed. Similarly for
+   code: please include the file, function/class, and line number(s) if that applies.
+
+4. Finally, _please keep issues focused on a single problem or suggestion_. If discussion prompts
+   you to think of a related issue, create a separate issue for that topic and add a link back to
+   the original discussion or issue (just use the "#NNN" syntax for issue/discussion/pull-request
+   NNN -- GitHub will automatically make this a link).
 
 
 Pull Requests
 --------------
 To contribute a change to the project, *please follow these steps*:
 
-  1. [Create a GitHub issue](https://github.com/RayTracing/raytracing.github.io/issues).
+  1. [Create a new GitHub issue](https://github.com/RayTracing/raytracing.github.io/issues).
 
-  2. Let us know whether you're willing to make the fix yourself.
+  2. Let us know if you're willing to make the fix yourself.
 
   3. Participate in the discussion as needed. We'll ensure that the work doesn't conflict with or
      duplicate other work planned or in progress, and decide which development branch is correct
      for the release type and release schedule.
 
-  4. Create your changes in a feature branch (or fork) from the assigned development branch (usually
-     `dev`).
+  4. Once you've received instructions to proceed with your change, create a new feature branch (or
+     fork) from the assigned development branch (usually `dev`).
 
   5. Follow the existing code style.
 
-  6. Include a one-line summary change at the bottom of the current development section in the
-     changelog. Include a reference to the associated GitHub issue.
+  6. Ensure that the change is complete:
+
+     - Update all relevant code for all three books (`src/*`). Since the code is developed as the
+       books proceed, you may need to update many historical code listings as well, _and this may
+       then require corresponding updates to the book text_.
+
+     - Update all relevant code listings and text in all three books (`books/RayTracing*.html`).
+       Follow existing style for the Markdeep source (for example, text should be wrapped to 100
+       characters).
+
+     - Provide clear and full commit descriptions: title line (50 characters max), followed by a
+       blank line, and then a descriptive body with lines not exceeding 72 characters. If a commit
+       is expected to completely resolve an outstanding issue, add a line "Resolves #NNN" to the
+       bottom of your commit message, where NNN is the existing GitHub issue number. You may provide
+       multiple such lines if applicable.
+
+     - Include a one-line summary change at the bottom of the current development section in the
+       changelog. Include a reference to the associated GitHub issue.
```

Comment:
Suggestion: Link to the Change log.

Reply:
I put the filename in parentheses. I don't want to link to a raw blob for a particular development branch, and the release version doesn't have the in-progress sections that the dev branches do. This should be sufficient, and I'm also encouraging interaction with the RTW developers instead of blind PRs anyway.